### ISSUE TITLE
running butch install stage1/2 a second time will rebuild gcc3

### ISSUE THREAD

**Issue by rofl0r** [sabotage-linux/sabotage#563]
```
root:/$ butch install stage1
...
package stage1 is already installed, skipping
*** downloadqueue ***
g++346
stage1-gcc3
*** buildqueue ***
g++346
stage1-gcc3
```

this happens because we `butch rm` the intermediate gccs in stage1.
maybe we should keep the entry in var/lib/butch.db to avoid this

**Comment by rofl0r** [comment]
i'm not really a fan of the calls to butch from stage1's body: it is actually just coincidence that it works, since the parent butch process might want to write to /var/lib/butch.db in parallel to the child processes, or even just writing its differing view of the database to the file.
maybe we should replace this hack with a script, which we can point out in README.
whatever we end up doing, an entry for stage1-gcc3 should be appended to butch.db after cleaning up. your thoughts, @JillTyrell ?

**Comment by JillTyrell** [comment]
I'm in favor of a script at this point since it can be used as the basis of a later more expansive script for stripping the build toolchain out of a finished chroot, which is something that would also be useful.

**Comment by JillTyrell** [comment]
Fixed via #588; safe to close.
